src/Ocelot/Values/ServiceHostAndPort.cs

```diff
@@ -1,20 +1,54 @@
-﻿namespace Ocelot.Values
+﻿namespace Ocelot.Values;
+
+public class ServiceHostAndPort : IEquatable<ServiceHostAndPort>
 {
-    public class ServiceHostAndPort
+    public ServiceHostAndPort(ServiceHostAndPort from)
     {
-        public ServiceHostAndPort(string downstreamHost, int downstreamPort)
-        {
-            DownstreamHost = downstreamHost?.Trim('/');
-            DownstreamPort = downstreamPort;
-        }
+        DownstreamHost = from.DownstreamHost;
+        DownstreamPort = from.DownstreamPort;
+        Scheme = from.Scheme;
+    }
 
-        public ServiceHostAndPort(string downstreamHost, int downstreamPort, string scheme)
-            : this(downstreamHost, downstreamPort) => Scheme = scheme;
+    public ServiceHostAndPort(string downstreamHost, int downstreamPort)
+    {
+        DownstreamHost = downstreamHost?.Trim('/');
+        DownstreamPort = downstreamPort;
+    }
 
-        public string DownstreamHost { get; }
+    public ServiceHostAndPort(string downstreamHost, int downstreamPort, string scheme)
+        : this(downstreamHost, downstreamPort) => Scheme = scheme;
 
-        public int DownstreamPort { get; }
-
-        public string Scheme { get; }
-    }
-}
+    public string DownstreamHost { get; }
+    public int DownstreamPort { get; }
+    public string Scheme { get; }
+
+    public override string ToString()
+        => $"{Scheme}:{DownstreamHost}:{DownstreamPort}";
+    public override int GetHashCode()
+        => Tuple.Create(Scheme, DownstreamHost, DownstreamPort).GetHashCode();
+
+    public bool Equals(ServiceHostAndPort other) => this == other;
+    public override bool Equals(object obj)
+        => obj != null && obj is ServiceHostAndPort o && this == o;

```

Comment:
```c#
public override bool Equals(object obj)
        => obj is ServiceHostAndPort o && this == o;
```
merge into pattern?

Reply:
> merge into pattern?

Swiss humor? Yeah! 🥲 I know you don't like my `Retry` pattern because you like Polly. 😏 
Let's ask another big fan of Polly... not @RaynaldM 🤣 ... and this is Mr. Seth @ks1990cn Hello Tanmay! 😉